controllers/kserve_inferenceservice_controller_test.go

```diff
@@ -87,11 +97,62 @@ var _ = Describe("The Openshift Kserve model controller", func() {
 				key := types.NamespacedName{Name: getKServeRouteName(inferenceService), Namespace: constants.IstioNamespace}
 				err = cli.Get(ctx, key, route)
 				return err
-			}, timeout, interval).ShouldNot(HaveOccurred())
+			}, timeout, interval).Should(Succeed())
 		})
+
+		It("should create required network policies when KServe is used", func() {
+			// given
+			inferenceService := &kservev1beta1.InferenceService{}
+			Expect(convertToStructuredResource(KserveInferenceServicePath1, inferenceService)).To(Succeed())
+			inferenceService.SetNamespace(testNs)
+
+			// when
+			Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+
+			// then
+			By("ensuring that the controller has created required network policies")
+			networkPolicies := &v1.NetworkPolicyList{}
+			Eventually(func() []v1.NetworkPolicy {
+				err := cli.List(ctx, networkPolicies, client.InNamespace(inferenceService.Namespace))
+				if err != nil {
+					Fail(err.Error())
+				}
+				return networkPolicies.Items
+			}, timeout, interval).Should(
+				ContainElements(
+					withMatchingNestedField("ObjectMeta.Name", Equal("allow-from-openshift-monitoring-ns")),
+					withMatchingNestedField("ObjectMeta.Name", Equal("allow-openshift-ingress")),
+					withMatchingNestedField("ObjectMeta.Name", Equal("allow-from-opendatahub-ns")),
+				),
```

Comment:
Are there other fields we need to check? Perhaps do a deep-equal check?

Reply:
Not sure how deeply we want to go with contracts here, I just wanted to make sure that we have created those policies. Based on my knowledge we are anyway not able to check in evntest if they actually work on a network level so my thinking was we can actually stop here. WDYT?

Reply:
FWIW,

IMHO, checking only for presence is just fine in this case, because it is almost a static NP. The only thing that varies is the namespace. If the NP would be different based on properties of something (for example, based on properties of the `InferenceService`), that would be worth additional assertions. But for these instances, I'm more than happy with just checking presence.

Reply:
No strong preference